bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
-
-	    // Uncomment the following to opt out of improving the service with this audio file
-	    //	    conn.setRequestProperty("X-WDC-PL-OPT-OUT", "1");
-            
+
+
+            // set up our http request
+            conn = setupHttpConnection(incomingBinaryPOSTData, serviceURL, credentials);
+
             // make the connection
             conn.connect();
-            
-            // send the POST request
-            OutputStream output = conn.getOutputStream();
-            output.write(incomingBinaryPOSTData);
-            output.flush();
-	    output.close();
-            
+
+            sendRequest(incomingBinaryPOSTData, conn);
+
             // Read the response from the service
-            BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), "UTF-8"));
+            String lines = readResponse(conn);
 
-	    if (DEBUG) {
-		System.out.println("Service response code: " + conn.getResponseCode());
-		System.out.println("Service response message: " + conn.getResponseMessage());
-	    }
+            // Parse the transcript string out of the Speech To Text service's JSON response
+            String transcript = readTranscripts(readJsonObject(new StringReader(lines)));
 
-            StringBuilder sb = new StringBuilder();
-            String line;
-            while ((line = br.readLine()) != null) {
-                sb.append(line + "\n");
-            }
-            br.close();
-
-	    if (DEBUG) System.out.println("Service response body:\n" + sb.toString());
-
-	    // Parse the transcript string out of the Speech To Text service's JSON response
-	    JsonReader responseReader = Json.createReader(new StringReader(sb.toString()));
-	    JsonObject jsonResponse = responseReader.readObject();
-	    responseReader.close();
-
-	    String transcript = "";
-	    try {
-		JsonArray resultsArray = jsonResponse.getJsonArray("results");
-		
-		for (JsonValue jv : resultsArray) {
-		    JsonObject jo = (JsonObject) jv;
-		    transcript = transcript + jo.getJsonArray("alternatives")
-			.getJsonObject(0)
-			.getString("transcript");
-		}
-	    } catch(Exception e) {
-		e.printStackTrace();
-		return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Unexpected response from service caused " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
-	    }
-            
             // Return the response from the service
-            return XML_DEC + "<results>" + STATUS_OK + "<transcript>" + transcript + "</transcript></results>";
-
-        } catch(Exception e){
+            return successXML(transcript);
+        } catch (Exception e) {
             e.printStackTrace();
-            return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Bluemix " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
-        }finally {
+            return failXML(e, "Bluemix ");
```

Comment:
What's the meaning of Bluemix here? Probably should be a constant, especially if it's used twice: here and on line 89.